dcpy/utils/s3.py

```diff
@@ -271,3 +280,41 @@ def get_file_as_stream(bucket: str, path: str) -> BytesIO:
 
 def get_file_as_text(bucket: str, path: str) -> str:
     return client().get_object(Bucket=bucket, Key=path).get("Body").read().decode()
+
+
+app = typer.Typer(add_completion=False)
+
+
+@app.command("upload_folder")
+def _cli_wrapper_upload_folder(
+    bucket: str = typer.Option(None, "-b", "--bucket", help="S3 bucket"),
+    local_path: Path = typer.Option(
+        None, "--folder-path", help="Path to local output folder"
+    ),
+    s3_path: Path = typer.Option(None, "--s3-path", help="Path to s3 output folder"),
+    acl: str = typer.Option(None, "-a", "--acl", help="Access level of file in s3"),
+    max_files: int = typer.Option(
+        20, "--max-files", help="Maximum number of files to upload"
+    ),
+    copy_contents: bool = typer.Option(
+        None,
+        "--copy-contents",
+        help="If true, uploads local folder into target folder. If false, uploads contents of local folder instead",
+    ),
+):
+    if copy_contents is not None:
+        include_foldername = not copy_contents
+    else:
+        include_foldername = None
```

Comment:
if the default for `copy_contents` were False rather than None, could it then just be passed to `upload_folder`?

Reply:
Oh that's a good point, this was clunky. Flipped the boolean logic of the argument to make it all cleaner as well